src/global/types.h

```diff
@@ -1933,14 +1941,6 @@ typedef struct REQUEST_INFO {
     int16_t x;
     int16_t y;
     uint16_t flags;
-    const char *heading_text;
-    char *item_texts;
-    int16_t item_text_len;
-    TEXTSTRING *heading;
-    TEXTSTRING *background;
-    TEXTSTRING *moreup;
-    TEXTSTRING *moredown;
-    TEXTSTRING *texts[MAX_REQLINES];
```

Comment:
Basically, on llvm/arm64, you want pointer members aligned to multiples of the 8-byte boundary. There's 3 ways to do this:

- Adding padding members, imho icky
- Doing weird tetris like having members that add up to multiples of 8 inbetween
- Putting all the pointers first

This seems the least icky?

Reply:
Sounds good, but what happens if they're not aligned? I think we have more structures like this BTW.

Reply:
All in all this seems to be pretty easy condition to break. I wonder if removing the `#pragma push(pack, 1)` from the beginning that forces the alignment would help? It's there mostly for legacy reason from the decompilation era. We should be good to remove it altogether for the most part, but I'm almost certain there are a couple of places left that use unsafe casts and rely on the member sizes/order. IMO it's better to track these and let the OS decide on data alignment for itself rather than fiddle with it by ourselves. Please LMK your thoughts.

Reply:
Yep it's a good point, I'll investigate what this macOS toolchain does if I remove the fence and the attrib, maybe it just ends up being happy by itself